Check/Extensions/Unrecommended.php

```diff
@@ -0,0 +1,144 @@
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('Not recommended');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Check for unrecommended modules.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {
+    $ret_val = dt('The following unrecommended modules(s) currently exist in your codebase: @list', array(
+      '@list' => implode(', ', array_keys($this->registry['extensions_unrec'])),
+    ));
+    if (drush_get_option('detail')) {
+      if (drush_get_option('html')) {
+        $ret_val .= '<br/>';
+        $ret_val .= '<table class="table table-condensed">';
+        $ret_val .= '<thead><tr><th>Name</th><th>Reason</th></thead>';
+        $ret_val .= '<tbody>';
+        foreach ($this->registry['extensions_unrec'] as $row) {
+          $ret_val .= '<tr><td>' . implode('</td><td>', $row) . '</td></tr>';
+        }
+        $ret_val .= '</tbody>';
+        $ret_val .= '</table>';
+      }
+      else {
+        foreach ($this->registry['extensions_unrec'] as $row) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 6);
+          }
+          $ret_val .= '- ' . $row[0] . ': ' . $row[1];
+        }
+      }
+    }
+    return $ret_val;
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('No unrecommended extensions were detected; no action required.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->getScore() != SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_PASS) {
+      return dt('Disable and completely remove unrecommended modules from your codebase for increased performance, stability and security in the any environment.');
+    }
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $this->registry['extensions_unrec'] = array();
+    $extension_info = $this->registry['extensions'];
+    uasort($extension_info, '_drush_pm_sort_extensions');
+    $unrecommended_extensions = $this->getExtensions();
+
+    foreach ($extension_info as $extension) {
+      $row = array();
+
+      $machineName = $extension->getName();
+
+      //get the human readable name of the module
+      if($extension->getType() == "module") {
+        $humanReadableName = \Drupal::moduleHandler()->getName($machineName);
+      }
+      //Cannot get the name of uninstalled themes. Needs to be fixed
```

Comment:
I am not able to figure out how to fix this. The problem arises because the `label` property of the extension is no longer provided by drush. So, I try to create the `label` using the machine readable and human readable name of the extension. But there is no way to get the human readable name of an uninstalled theme.